Submission soft-delete functionality added

ihsaan-ullah opened the pull request:
# @ mention of reviewers
@adiprerepa 


# A brief description of the purpose of the changes contained in this PR.
...


# Issues this PR resolves
...



# A checklist for hand testing
- [ ] add checklist here


# Any relevant files for testing
[link]('#') to any relevant files (or drag and drop into github)


# Checklist
- [ ] Code review by me 
- [ ] Hand tested by me 
- [ ] I'm proud of my work
- [ ] Code review by reviewer
- [ ] Hand tested by reviewer
- [ ] CircleCi tests are passing
- [ ] Ready to merge

Didayolo commented:
@ihsaan-ullah 

I think the deletion button should only appear in the resources interface:

<img width="1197" alt="Capture d’écran 2025-02-04 à 16 26 07" src="https://github.com/user-attachments/assets/9633e73c-4373-48d3-be47-c472264fc6e7" />


and not in the submissions table from the competition:

<img width="1258" alt="Capture d’écran 2025-02-04 à 16 26 16" src="https://github.com/user-attachments/assets/f2dae43e-e180-4b05-bb0f-c8c712812918" />


By the way, it looks like only new submissions (after deployment of this feature) are possible to delete:

<img width="815" alt="Capture d’écran 2025-02-04 à 16 28 38" src="https://github.com/user-attachments/assets/4b20c049-3dc5-400d-b233-ff2bde9b9127" />

Didayolo commented:
OK, works for me. I think the deletion from resources interface is not working then, only from the competition page.

Didayolo commented:
- Updating the error message on resources interface (you can go to the competition etc.)
- Testing big files to see if you save the quota

ihsaan-ullah commented:
@ObadaS this is interesting. I will check on my side with a big file to see why full space is not freed.

ihsaan-ullah commented:
@ObadaS I checked on my side and I think when a submission is processed, some additional files are stored somewhere that takes additional space. Below are the details of my test

- No file upload
    Quota: 0.0B / 15.0GiB
- Uploaded Iris bundle:
    Quota: 907.1KiB / 15.0GiB
- Uploaded big submission (zip size: 1.34 GB)
    Quota: 1.3GiB / 15.0GiB     --- (1.40 GB)
- Uploaded big submission (zip size: 1.34 GB)
    Quota: 2.5GiB / 15.0GiB     --- (2.68 GB)
- When both submissions ran successfully
    Quota: 5.0GiB / 15.0GiB
- Deleted submission 1
    Quota: 3.8GiB / 15.0GiB
- Deleted submission 2
    Quota: 2.5GiB / 15.0GiB

ObadaS commented:
@ihsaan-ullah I think you can get more information about the files that are stored and created by looking at the logs of the compute_worker. It seems like, after downloading the file and doing the work needed, it generates a new file that gets uploaded to the storage. If I remember well, I read something about predictions files, since the ingestion and scoring program can be different, the generated files needs to be uploaded to the storage ?
I will look into it later and try to get logs about those mysterious files.

I agree about changing the size unit to GB to be less confusing since people are more used to this. If not too complicated, it would also be nice if the size change dynamically (shows in MB or KB if needed, instead of showing 0GB or 0.1GB)

ihsaan-ullah commented:
> I agree about changing the size unit to GB to be less confusing since people are more used to this. If not too complicated, it would also be nice if the size change dynamically (shows in MB or KB if needed, instead of showing 0GB or 0.1GB)

Sounds good. I will check this

ihsaan-ullah commented:
@ObadaS @Didayolo 

### I made the following updates:
- size formatter functions in both python and js updated to calculate size in GB/MB/KB instead of GiB/MiB/KiB 
- updated soft-deletion to fix the problem with remaining space that was left even after deleting everything
- changed default user quota size from 15 GiB to 15 GB
- updated user quota calculation to use GB/MB/KB instead of GiB/MiB/KiB 

### Some problems/thoughts
- In the calculation of storage analytics I found the use of GiB/MiB/KiB. Maybe there are more places where we have such inconsistency. I suggest that we convert this point into a separate issue and fix it later.
- I found redundant fields: `prediction_result_file_size`, `scoring_result_file_size`, `detailed_result_file_size`. These only keep the size of the files while the file fileds are also present and we can take the size from the file fields directly. We can ignore this point or we clean this up later in a separate issue/PR

Didayolo commented:
@ihsaan-ullah I confirm that now deleting submission does decrease the quota accordingly.

I've just seen something a bit shady.

I uploaded a submission with a size of 442 MB: 

<img width="400" alt="Capture d’écran 2025-02-10 à 16 56 29" src="https://github.com/user-attachments/assets/46100dea-84ef-465f-976f-1fc725ab537a" />

When the submission was being processed, I could see a consistent quota:

<img width="245" alt="Capture d’écran 2025-02-10 à 16 56 42" src="https://github.com/user-attachments/assets/9b1c12b4-67e4-4869-aa3f-fd1db163df3e" />

But at the end of the computation, the quota used by the submission doubled:

<img width="273" alt="Capture d’écran 2025-02-10 à 16 56 24" src="https://github.com/user-attachments/assets/ffca60fe-a1f4-42ed-b1d5-f1921d0252b6" />

Is that normal?

ihsaan-ullah commented:
> @ihsaan-ullah I confirm that now deleting submission does decrease the quota accordingly.
> 
> I've just seen something a bit shady.
> 
> I uploaded a submission with a size of 442 MB:
> 
> <img alt="Capture d’écran 2025-02-10 à 16 56 29" width="400" src="https://private-user-images.githubusercontent.com/11784999/411638736-46100dea-84ef-465f-976f-1fc725ab537a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMDM1NDcsIm5iZiI6MTczOTIwMzI0NywicGF0aCI6Ii8xMTc4NDk5OS80MTE2Mzg3MzYtNDYxMDBkZWEtODRlZi00NjVmLTk3NmYtMWZjNzI1YWI1MzdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDE2MDA0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYzMmY3YTQxNTQ5NWU3YzBiOTk0YTZiMzUzMzAxMDBiNmFjN2FjNWNiMWQ2ODE0OTMxZDczN2E1YTVhNzUxY2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dae_3MqTy6hgL_q2gGZbpUAiij0lpm6mFX0_u2b2Ma4">
> When the submission was being processed, I could see a consistent quota:
> 
> <img alt="Capture d’écran 2025-02-10 à 16 56 42" width="245" src="https://private-user-images.githubusercontent.com/11784999/411638742-9b1c12b4-67e4-4869-aa3f-fd1db163df3e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMDM1NDcsIm5iZiI6MTczOTIwMzI0NywicGF0aCI6Ii8xMTc4NDk5OS80MTE2Mzg3NDItOWIxYzEyYjQtNjdlNC00ODY5LWFhM2YtZmQxZGIxNjNkZjNlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDE2MDA0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg5YzJmOWMyNmY5ODI4MmQyYzRiYjhmY2M4MTQwZmI2NjM3MjE1YzQ2ODJiYTgyMzI2MjFkYjgyNzdlOTc0MDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.x2soQsYif1aBn4yZDrd2FP71TYy64xn_9xhcDD1c9qo">
> But at the end of the computation, the quota used by the submission doubled:
> 
> <img alt="Capture d’écran 2025-02-10 à 16 56 24" width="273" src="https://private-user-images.githubusercontent.com/11784999/411638724-ffca60fe-a1f4-42ed-b1d5-f1921d0252b6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzkyMDM1NDcsIm5iZiI6MTczOTIwMzI0NywicGF0aCI6Ii8xMTc4NDk5OS80MTE2Mzg3MjQtZmZjYTYwZmUtYTFmNC00MmVkLWIxZDUtZjE5MjFkMDI1MmI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjEwVDE2MDA0N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YzNmOTcwNWY5OTk3ODA0OTM5NTc0MzFlOWUwNDdmZWI4YmIzODNjYWNkNWJjNmM2MTg5ZmZkY2ZmMzMwOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.1m3zDbRH-BHnC-h63vmU-iQ2BaJRHwD3Yy8jK9kbbUI">
> Is that normal?

This needs a bit investigation but I think this comes from the prediction files stored after the submission is processed. If you remember, the prediction zip also includes the submission itself. BUT we should look into it together with my other points in my last comment
